PhysicsTools/PatAlgos/plugins/PATMuonSlimmer.cc

```diff
@@ -111,6 +118,58 @@ pat::PATMuonSlimmer::produce(edm::Event & iEvent, const edm::EventSetup & iSetup
 	if (linkToPackedPF_) {
             mu.refToOrig_ = refToPtr(mu2pc[mu.refToOrig_]);
         }
+        if (dropDirectionalIso_(mu)) {
+            reco::MuonPFIsolation zero;
+            mu.setPFIsolation("pfIsoMeanDRProfileR03",zero);
+            mu.setPFIsolation("pfIsoSumDRProfileR03",zero);
+            mu.setPFIsolation("pfIsoMeanDRProfileR04",zero);
+            mu.setPFIsolation("pfIsoSumDRProfileR04",zero);
+        }
+        if (dropPfP4_(mu)) mu.setPFP4(reco::Particle::LorentzVector());
```

Comment:
actually, we can do away with the interface change in reco::Muon and just add ``` && mu.isPFMuon ``` in the conditional.